### ISSUE TITLE
Add Citation type

### ISSUE THREAD

**Issue by nokome** [stencila/schema#90]
Need something similar to Pandoc's `Citation` element:
https://github.com/jgm/pandoc-types/blob/1.17.5.4/Text/Pandoc/Definition.hs#L276-L283 to represent an in-prose reference to a `CreativeWork`.

**Comment by nokome** [comment]
Just to transfer some notes from elsewhere...

There are various ways that citations are represented including,

- As a superscript after some test e.g. `17, 18`
- In brackets after some text e.g.: `[17]–[18]`
- Names and years in parentheses e.g. `(Smith 1990, Jones et al 2018)`

Pandoc appears to capture these alternatives in a `CitationMode` type (from above link):

```
data Citation = Citation { citationId      :: String
                         , citationPrefix  :: [Inline]
                         , citationSuffix  :: [Inline]
                         , citationMode    :: CitationMode
                         , citationNoteNum :: Int
                         , citationHash    :: Int
                         }

data CitationMode = AuthorInText | SuppressAuthor | NormalCitation
```

We want to try and keep nodes as semantic as possible and avoid encoding presentational choices into the schema. Instead, ideally, different themes should be able to take the encoded HTML (with `itemprop`s) and style it as they want.

**Comment by nokome** [comment]
Closed by #121
